### DISCUSSION TITLE
S3 client-side encryption legacy error

### DISCUSSION THREAD

**Discussion by Davy1992** [aws/aws-sdk-php#2509] [General]
I have an issue with this code when trying to decrypt a SES encrypted email with client-side encryption.
https://github.com/aws/aws-sdk-php/blob/0b9277a5c3d2fbccafc16b05a1b53404c9b67d1b/src/S3/Crypto/S3EncryptionClientV2.php#L338-L349
The method I am calling is **getObjectAsync()** which promises to return an object from S3 and decrypt the data in the 'Body' field.
Instead, I get an error message triggered by the code above.
Even though I **Explicitly** defined the `'@SecurityProfile' => 'V2_AND_LEGACY'`

In this scenario what am I supposed to do to prevent this error from stopping my code and decrypting the message.
Is there a way to make SES use the V2 encryption mode?

FYI: I am using the [example code](https://docs.aws.amazon.com/aws-sdk-php/v3/api/class-Aws.S3.Crypto.S3EncryptionClientV2.html)

**Comment by yagodinez** [comment]
Got into the same issue. A possible solution is to disable the warning at runtime.

```php
// Report all errors except E_WARNING (deprecation notices)
error_reporting(E_ERROR & ~E_WARNING);
```
